### PR TITLE
Support json_extract function

### PR DESCRIPTION
**Pull request by mooleetzi**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [X] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #4644 

## What this PR does / why we need it:
Support query function for json type

### CODE REVIEW

**Comment by mooleetzi** [comment]
I transferred json_extract to jq, but CI told me I need to add json_extract  😿

**Comment by yingfeng** [comment]
> I transferred json_extract to jq, but CI told me I need to add json_extract  😿

@guguducken what does this mean?  since `github.com/matrixorigin/matrixone/pkg/vectorize/json_extract` is deleted in this PR
```
  cd $GITHUB_WORKSPACE/matrixone
  echo "-----------------------ut start-----------------------"
  CGO_CFLAGS="-I${GITHUB_WORKSPACE}/matrixone/cgo"
  CGO_LDFLAGS="-L${GITHUB_WORKSPACE}/matrixone/cgo -lmo -lm"
  go test -v -cover -tags matrixone_test -p 1 -covermode=count --count=1 github.com/matrixorigin/matrixone/pkg/container/bytejson github.com/matrixorigin/matrixone/pkg/container/types github.com/matrixorigin/matrixone/pkg/errno github.com/matrixorigin/matrixone/pkg/sql/parsers/dialect/mysql github.com/matrixorigin/matrixone/pkg/sql/plan/function/builtin/binary github.com/matrixorigin/matrixone/pkg/sql/plan/function github.com/matrixorigin/matrixone/pkg/vectorize/jq github.com/matrixorigin/matrixone/pkg/vectorize/json_extract | tee 'UT-Report.out'
  cat 'UT-Report.out' | grep -a -P "(?m)(?<=ok|\?).*(?=%|])" > 'ut_coverage.raw'
  echo "-----------------------ut finish-----------------------"
  shell: /usr/bin/bash -e {0}
  env:
    raw_ut_coverage: 'ut_coverage.raw'
    ut_report: 'UT-Report.out'
-----------------------ut start-----------------------
no required module provides package github.com/matrixorigin/matrixone/pkg/vectorize/json_extract; to add it:
	go get github.com/matrixorigin/matrixone/pkg/vectorize/json_extract
Error: Process completed with exit code 1.

```

**Comment by guguducken** [comment]
> > I transferred json_extract to jq, but CI told me I need to add json_extract  😿
> 
> @guguducken what does this mean? since `github.com/matrixorigin/matrixone/pkg/vectorize/json_extract` is deleted in this PR
> 
> ```
>   cd $GITHUB_WORKSPACE/matrixone
>   echo "-----------------------ut start-----------------------"
>   CGO_CFLAGS="-I${GITHUB_WORKSPACE}/matrixone/cgo"
>   CGO_LDFLAGS="-L${GITHUB_WORKSPACE}/matrixone/cgo -lmo -lm"
>   go test -v -cover -tags matrixone_test -p 1 -covermode=count --count=1 github.com/matrixorigin/matrixone/pkg/container/bytejson github.com/matrixorigin/matrixone/pkg/container/types github.com/matrixorigin/matrixone/pkg/errno github.com/matrixorigin/matrixone/pkg/sql/parsers/dialect/mysql github.com/matrixorigin/matrixone/pkg/sql/plan/function/builtin/binary github.com/matrixorigin/matrixone/pkg/sql/plan/function github.com/matrixorigin/matrixone/pkg/vectorize/jq github.com/matrixorigin/matrixone/pkg/vectorize/json_extract | tee 'UT-Report.out'
>   cat 'UT-Report.out' | grep -a -P "(?m)(?<=ok|\?).*(?=%|])" > 'ut_coverage.raw'
>   echo "-----------------------ut finish-----------------------"
>   shell: /usr/bin/bash -e {0}
>   env:
>     raw_ut_coverage: 'ut_coverage.raw'
>     ut_report: 'UT-Report.out'
> -----------------------ut start-----------------------
> no required module provides package github.com/matrixorigin/matrixone/pkg/vectorize/json_extract; to add it:
> 	go get github.com/matrixorigin/matrixone/pkg/vectorize/json_extract
> Error: Process completed with exit code 1.
> ```

This is because this unit test with modified file path did not think of the folder deleted scenario, I will fix it

**Comment by guguducken** [comment]
@matrix-meow rerun failed

**Review by fengttt — CHANGES_REQUESTED** [review]
Is there a document on the jq syntax we will support?

**Comment by mooleetzi** [comment]
> Is there a document on the jq syntax we will support?

I'll add them later, Thx.

**Comment by guguducken** [comment]
> > > @matrix-meow rerun failed
> > 
> > 
> > All failed jobs are rerun ----- @matrix-meow
> 
> Hello, CI still let me add `github.com/matrixorigin/matrixone/pkg/vectorize/json_extract`.

Hello, I will fix this tomorrow, This CI does not affect the merge at this time.

**Comment by mooleetzi** [comment]
> > > > @matrix-meow rerun failed
> > > 
> > > 
> > > All failed jobs are rerun ----- @matrix-meow
> > 
> > 
> > Hello, CI still let me add `github.com/matrixorigin/matrixone/pkg/vectorize/json_extract`.
> 
> Hello, I will fix this tomorrow, This CI does not affect the merge at this time.

🫡

**Comment by fengttt** [comment]
> > Is there a document on the jq syntax we will support?
> 
> I'll add them later, Thx.

This change won't be approved until we see the doc.

**Review by ouyuanning — APPROVED** [review]

File: pkg/sql/parsers/dialect/mysql/mysql_sql.y [hunk 1]
```diff
@@ -5466,7 +5466,7 @@ function_call_generic:
         }
 	}
 function_call_json:
-	JSON_EXTRACT '(' STRING ',' STRING ')'
+	JQ '(' STRING ',' STRING ')'
```
Comment (iamlinjunhong) [inline pkg/sql/parsers/dialect/mysql/mysql_sql.y:5469]: add unit test in mysql_sql_test.go

**Review by fengttt — APPROVED** [review]
Rename jq to extract, this is good to go.

Thank you very much,

File: pkg/sql/parsers/dialect/mysql/mysql_sql.y (hunk 1 above)
Comment (mooleetzi) [reply pkg/sql/parsers/dialect/mysql/mysql_sql.y:5486]: OK，I will do it right away

**Comment by mooleetzi** [comment]
> Rename jq to extract, this is good to go.
> 
> Thank you very much,

Ok. Thank you too.

**Pull request merged**